pandas/core/indexes/datetimes.py

```diff
@@ -787,8 +787,7 @@ def snap(self, freq='S'):
             snapped[i] = s
 
         # we know it conforms; skip check
-        return DatetimeIndex._simple_new(snapped, freq=freq)
-        # TODO: what about self.name?  tz? if so, use shallow_copy?
```

Comment:
Can you also address the `tz` portion of the TODO, as that also appears to not be preserved?

Reply:
Is there a reason we couldn't address that issue here (in a separate commit) ?

Reply:
@gfyoung : a little unclear on my part, but that's what I meant; I think it should be addressed in this PR as I think the fix should be the same as with `name` and just be a matter of adding `tz=self.tz` to the return.

Reply:
Fixed. Thanks for point the `tz` out to me!